parsers/US_CA.py

```diff
@@ -26,159 +56,91 @@ def fetch_production(
-        "Solar": "solar",
-        "Wind": "wind",
-        "Geothermal": "geothermal",
-        "Biomass": "biomass",
-        "Biogas": "biomass",
-        "Small hydro": "hydro",
-        "Coal": "coal",
-        "Nuclear": "nuclear",
-        "Natural Gas": "gas",
-        "Large Hydro": "hydro",
-        "Other": "unknown",
-    }
-    storage_map = {"Batteries": "battery"}
-    daily_data = []
-    for i in range(0, latest_index + 1):
-        h, m = map(int, csv["Time"][i].split(":"))
-        date = (
-            arrow.utcnow()
-            .to("US/Pacific")
-            .replace(hour=h, minute=m, second=0, microsecond=0)
-        )
+    csv = pandas.read_csv(target_url)
+
+    # Filter out last row if timestamp is 00:00
+    if  csv.iloc[-1]["Time"] =="OO:OO":
+        df = csv.copy().iloc[:-1]
+    else:
+        df = csv.copy()
+    df = df.rename(columns={**PRODUCTION_MODES_MAPPING, **STORAGE_MAPPING})
+    df = df.groupby(level=0, axis=1).sum()
+
+    all_data_points = []
+    for index, row in df.iterrows():
+        production = {}
+        storage = {}
+
+        row_datetime = target_datetime.replace(hour= int(row["Time"][:2]), minute=int(row["Time"][-2:]))
+        for mode in [mode for mode in PRODUCTION_MODES if mode in row]:
+            production_value =  float(row[mode])
+            if production_value < 0 and (
+                mode in  ["solar", "nuclear", "coal","gas","unknown","biomass"]
+            ):
+                logger.warn(
+                    f" {mode} production for US_CA was reported as less than 0 and was clamped"
+                )
+                production_value = 0.0
+
+            production[mode] = production_value
+
+        storage["battery"] = float(row["battery"])
+
         data = {
             "zoneKey": zone_key,
-            "production": defaultdict(float),
-            "storage": defaultdict(float),
+            "production": production,
+            "storage": storage,
             "source": "caiso.com",
-            "datetime": date.datetime,
+            "datetime": arrow.get(row_datetime).replace(tzinfo="US/Pacific").datetime,
         }
+        all_data_points.append(data)
+    return all_data_points
 
-        # map items from names in CAISO CSV to names used in Electricity Map
-        for ca_gen_type, mapped_gen_type in production_map.items():
-            production = float(csv[ca_gen_type][i])
-
-            if production < 0 and (
-                mapped_gen_type == "solar" or mapped_gen_type == "nuclear"
-            ):
-                logger.warn(
-                    ca_gen_type
-                    + " production for US_CA was reported as less than 0 and was clamped"
-                )
-                production = 0.0
-
-            # if another mean of production created a value, sum them up
-            data["production"][mapped_gen_type] += production
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: str = "US-CA",
```

Comment:
```suggestion
    zone_key: str = "US-CAL-CISO",
```
I'm not even sure there should be a default key here.